euth/rates/serializers.py

```diff
@@ -1,15 +1,42 @@
+from django.contrib.contenttypes.models import ContentType
 from rest_framework import serializers
 
+from euth.rates import models as rate_models
+
 from .models import Rate
 
 
 class RateSerializer(serializers.ModelSerializer):
-    user_name = serializers.SerializerMethodField()
+    meta_info = serializers.SerializerMethodField()
 
     class Meta:
         model = Rate
-        read_only_fields = ('id', 'user_name')
+        read_only_fields = ('id', 'meta_info')
         exclude = ('user', 'modified', 'created')
 
-    def get_user_name(self, obj):
-        return str(obj.user.username)
+    def get_meta_info(self, obj):
```

Comment:
I agree that it would be better to have this code in at the model. If you need any additional data, you can pass it as arguments to the method.

```python
class Rate(models.Model):
    POSITIVE = 1
    NEGATIVE = -1

    def get_meta_info(self, user):
        contenttype = ContentType.objects.get_for_model(self.content_object)
        rates = rate_models.Rate.objects.filter(
            content_type=contenttype,
            object_pk=self.content_object.pk
        )
        result['positive_rates_on_same_object'] = rates.filter(value=self.POSITIVE).count()
        result['negative_rates_on_same_object'] = rates.filter(value=self.NEGATIVE).count()

        try:
            user_rate_on_same_object = rates.get(user=user)
            result['user_rate_on_same_object_value'] = user_rate_on_same_object.value
            result['user_rate_on_same_object_id'] = user_rate_on_same_object.pk
        except:
            result['user_rate_on_same_object_value'] = None
            result['user_rate_on_same_object_id'] = None

        return result
```

Furthermore `1` and `-1` should be symbols, so that they defined at a single place. Also you should add an expression to `except` so that it only catches the exceptions you want to handle.

Please see Tom Christie's article [Django models, encapsulation and data integrity](https://www.dabapps.com/blog/django-models-and-encapsulation/) for more information about this concept.